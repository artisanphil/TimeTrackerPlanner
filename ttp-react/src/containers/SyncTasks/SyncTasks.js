import React, { Component } from 'react';

const API = 'https://localhost:5001/api/';

class SyncTasks extends Component {
    constructor(props) {
      super(props);
  
      // This binding is necessary to make `this` work in the callback
      this.handleClick = this.handleClick.bind(this);
    }
  
    handleClick() {
        console.log("clicked sync");
        
        //var tasks = {"planid":493,"starttime":1545652127375,"duration":0};
        var tasks = '{"taskDone":' + localStorage.getItem("tasks") + '}';

        console.log(API + 'plannedtasks');
        console.log(tasks);
        fetch(API + 'plannedtasks',
        {
            method: 'POST',
            headers: {
                  'Accept': 'application/json',
                  'Content-Type': 'application/json'
            },
            body: tasks
        }).then(function(response) {
          console.log(response);
        })
        .catch(function(error) {
          console.log(error);
      });   
    }

    render() {
      return (
        <button onClick={this.handleClick}>
            Sync 
        </button>
      );
    }
  }

  export default SyncTasks;