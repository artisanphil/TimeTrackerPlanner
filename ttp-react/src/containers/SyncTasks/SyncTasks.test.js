import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SyncTasks from './SyncTasks';

describe('SyncTasks', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SyncTasks />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <SyncTasks />
    );
    
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});