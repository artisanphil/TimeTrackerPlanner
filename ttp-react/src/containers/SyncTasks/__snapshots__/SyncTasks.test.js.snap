// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncTasks has a valid snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Sync
  </button>
  <div>
    
  </div>
</div>
`;
